sklearn/preprocessing/imputation.py

```diff
@@ -375,3 +375,203 @@ def transform(self, X):
+        """
+        self._validate_params()
+
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64)
+        self.n_features_ = X.shape[1]
+
+        if (isinstance(self.features, six.string_types) and
+                self.features == "train"):
+            _, self.feat_with_missing_ = self._get_missing_features_info(X)
+
+        return self
+
+    def transform(self, X):
+        """Generate missing values indicator for X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            The input data to complete.
+
+        Returns
+        -------
+        Xt : array or sparse matrix, shape = [n_samples, n_features]
+             The missing indicator for input data
+
+        """
+        if (isinstance(self.features, six.string_types) and
+                self.features == "train"):
+            check_is_fitted(self, "feat_with_missing_", "n_features_")
+        else:
+            check_is_fitted(self, 'n_features_')
+
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64)
+        if X.shape[1] != self.n_features_:
+            raise ValueError("X has a different number of features "
+                             "than during fitting.")
+
+        imputer_mask, feat_with_missing = self._get_missing_features_info(X)
+
+        if isinstance(self.features, six.string_types):
+            if self.features == "train":
+                features = np.setdiff1d(feat_with_missing,
+                                        self.feat_with_missing_)
+                if features.size > 0:
+                    warnings.warn("The features %s have missing values "
+                                  "in transform but have no missing values "
+                                  "in fit " % features, RuntimeWarning,
+                                  stacklevel=1)
+                imputer_mask = imputer_mask[:, self.feat_with_missing_]
+
+        else:  # features is array-like
+            imputer_mask = imputer_mask[:, self.features]
+
+        return imputer_mask
+
+    def fit_transform(self, X, y=None):
+        """Generate missing values indicator for X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            The input data to complete.
+
+        Returns
+        -------
+        Xt : array or sparse matrix, shape = [n_samples, n_features]
+             The missing indicator for input data
+
+        """
+        self._validate_params()
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64)
+
+        imputer_mask, feat_with_missing = self._get_missing_features_info(X)
+
+        self.n_features_ = X.shape[1]
+        if isinstance(self.features, six.string_types):
+            if self.features == "train":
+                self.feat_with_missing_ = feat_with_missing
+                imputer_mask = imputer_mask[:, self.feat_with_missing_]
```

Comment:
I believe this is the line causing the failure in old scipy. Meanwhile I will address your other comments.